handler/handler.go

```diff
@@ -26,7 +27,13 @@ func CreateNatsQueue(address string, port int) (*NatsQueue, error) {
 	clientID := "faas-publisher-" + val
 	clusterID := "faas-cluster"
 
-	nc, err := stan.Connect(clusterID, clientID, stan.NatsURL(natsURL))
+	conn, err := nats.Connect(natsURL, nats.ReconnectHandler(queue1.reconnectClient(clientID, clusterID)))
```

Comment:
Please can you explain why this changed from `stan` to `nats` and what the implications are?

Reply:
In order to to define a `ReconnectHandler`, it must be set directly on the NATS connection.  The NATS streaming connection (created by the `stan` package) accepts an existing NATS connection as a parameter so I created a connection with a `ReconnectHandler` and passed into `stan`.

Alternatively, it should be possible to create a connection through `stan`, then set the `ReconnectHandler` on the underlying NATS Streaming connection with `nc.NatsConn().SetReconnectHandler`

After looking deeper into it, the difference here is we are now responsible for closing the NATS connection.

Reply:
The NATS client has internal reconnection logic already, so I think maybe using the `MaxReconnects` option might fit better here?  When connecting to NATS Streaming via `stan.Connect` internally the STAN client will use the default reconnection logic which is to attempt to connect a few times to the server (reconnecting state) before giving up entirely and closing the connection (closed state which is terminal so client will not reconnect again).
By setting the `ReconnectHandler` and `ClosedHandler` it is possible to add custom logic or logging for after the reconnection or closing of the connection has occurred (for example could make the server exit when closed state is reached meaning that client will not reconnect to NATS again).
It is also possible to make the client reconnect forever rather than giving up by setting `nats.MaxReconnects(-1)` in case that behavior might be better suited too.

Reply:
Hey @wallyqs.  The underlying NATS connection reconnect as you described. The issue is the NATS Streaming server loses all of it's data because it is in memory mode.  Part of this is the client IDs.  Because of this a new streaming connection has to be made to rejoin the cluster and register itself.

Btw I am working on restructuring this commit to be more reviewer friendly.

Reply:
Ok might be missing something then... The reconnect handler is called when the NATS client has successfully established a connection to a new server and the NATS client internally replays all the subscriptions once having reconnected, so for the STAN client the reconnection to NATS would be transparent.

Reply:
@wallyqs , unfortunately it hasn't worked that way in my experience, and it is documented in https://github.com/nats-io/nats-streaming-server/issues/150 .  I agree with you that the STAN client should resubscribe, and this may might be an issue that could be addressed by that library.  But they don't seem to mind (https://github.com/nats-io/go-nats-streaming/issues/153)